### PR TITLE
🚧 Resolves #5, resolves #6 create a browser compatible version and use standard project structure

### LINKED ISSUE #5: Create a browser compatible version
We can use https://webpack.js.org/ to create a browser compatible version of this library.
As you said, `plantuml-encoder` is browser compatible so we just need to extract all the Node specific code into a dedicated file.

### LINKED ISSUE #6: Use a "standard" structure for the project
I think we should use a "standard" structure for the project. It will make it easier to configure tools in the JavaScript ecosystem.

Something like:

```sh
asciidoctor-plantuml.js
  |- package.json
  |- /dist
  |- /src
    |- index.js
  |- /spec
    |- asciidoctor-plantuml.spec.js
  |- /examples
    |- convert-fetch.js
    |- convert.js
    |- plantuml.adoc
```

### PR DESCRIPTION
**Pull request by Mogztter**
Based on #6

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const webpack = require('webpack');
+const path = require('path');
+
+const config = {
+  entry: './asciidoctor-plantuml.js',
+  output: {
+    path: path.resolve(__dirname, 'dist'),
+    filename: 'bundle.js',
+    library: 'AsciidoctorPlantuml'
```
Comment (Mogztter) [inline webpack.config.js]: The extension will be available in a browser, as a global variable `AsciidoctorPlantuml`.

**Comment by Mogztter** [comment]
> Why asciidoctor-plantuml.js is not moved to either lib or src ?

`asciidoctor-plantuml.js` is the entry point (usually this file is named `main.js` and contains a single `require`).
Since you said that you wanted to keep the name `asciidoctor-plantuml.js`, it will be a bit odd to have:

**asciidoctor-plantum.js**
```js
require('./lib/asciidoctor-plantum.js')
```

If we want to move all the code to `lib` then we should rename `asciidoctor-plantuml.js` to `main.js` and create a file named `asciidoctor-plantuml.js` in `lib` with all the code.
What do you think ?

> Is there any possibility to reuse code from node test in browser test ? Doing this we could reuse the same test suite across 2 implementations having better coverage

Actually I did try to do it but Webpack was failing on cheerio. So in theory it's possible but in practice that would require to refactor the code to hide/remove some dependencies (because we have to "browserify" the tests).
The simplest solution is to avoid using Node modules in the test suites so it's straight forward to run them in a browser environment.

**Comment by eshepelyuk** [comment]
1. I assumed the whole `lib` will be copied to `dist`, so that `lib` will not be in the require path

1. Since cheerio claims to mimic jquery - we can conditionally require either cheerio or jquery in jasmine spec. That's my assumption. I'm really worried by the case of having single test case for browser version and the need to maintain two separate test suite while potentially test codebase may be reused.

**Comment by Mogztter** [comment]
>I assumed the whole lib will be copied to dist, so that lib will not be in the require path

Files are not exactly copied but "compiled". Webpack will resolve the dependencies, create a standalone bundle and export the library as a global variable.
But yes, `lib` will not be in the require path anymore because Webpack will rewrite the require directive.

> Since cheerio claims to mimic jquery - we can conditionally require either cheerio or jquery in jasmine spec. That's my assumption. I'm really worried by the case of having single test case for browser version and the need to maintain two separate test suite while potentially test codebase may be reused.

Be assured this is the short term solution to avoid a major refactoring but we do need to work on this issue 👍

**Comment by Mogztter** [comment]
The code base is now shared between the Node.js tests suite and the browser tests suite.
Some tests are not shared because they were either using `process.env.PLANTUML_SERVER_URL` or the file system (`fs`, `path` and/or `tmp` modules).

I was forced to use browserify to resolve all the dependencies in the browser tests suite.

**Pull request merged**